cmake/PFUNITConfig.cmake.in

```diff
@@ -12,14 +12,36 @@ include(CMakeFindDependencyMacro)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}")
 
+# Enables the use of the build directory directly, without having to
+# specify the paths for the external packages as well
 if (EXISTS "@CMAKE_BINARY_DIR@/extern/gFTL")
   list(APPEND CMAKE_PREFIX_PATH "@CMAKE_BINARY_DIR@/extern/gFTL")
+elseif (EXISTS "@GFTL_ROOT@")
+  set(GFTL_ROOT "@GFTL_ROOT@")
+elseif (EXISTS "@GFTL_DIR@")
+  set(GFTL_ROOT "@GFTL_DIR@")
+elseif (EXISTS "@GFTL_TOP_DIR@")
+  set(GFTL_ROOT "@GFTL_TOP_DIR@")
+endif()
```

Comment:
This feels a little bit gross, but basically covers all the ways pFUnit could have been configured with gFTL: directly via an installation, directly via a build directory, indirectly via fArgParse

Reply:
You probably do not need to support that last one.     I'm more concerned that the build accidentally get conflicts because gFTL is built inside in more than one location.  (And this has happened in the past.)

Reply:
Yes, exactly. These changes, plus those in the `build_submodule` CMake function should prevent this, as well as ordering the calls to find fArgParse first. This should mean that the exact version/build of gFTL is propagated through gFTL-shared, fArgParse and finally up to pFUnit.

That's the idea at least! I've tested "out of the box" build and installation, using `FetchContent`, using an fArgParse installation, using a gFTL installation to build fArgParse and building pFUnit against that.

...and I've just spotted a problem, when using the submodules, it tries to build them directly in the source directories. I need to fix that

Reply:
Ok that issue is fixed. I've made the change in the rest of the repos too. The submodules don't need to be bumped, as they end up using the `build_submodule` from this repo.